Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -314,11 +314,48 @@ extension StripeCardReaderService: CardReaderService {
             .eraseToAnyPublisher()
     }
 
+    public func retryActivePaymentIntent() -> AnyPublisher<PaymentIntent, Error> {
+        guard let activePaymentIntent = activePaymentIntent else {
+            return Fail(error: CardReaderServiceError.retryNotPossibleNoActivePayment)
+                .eraseToAnyPublisher()
+        }
+        switch activePaymentIntent.status {
+        case .requiresPaymentMethod:
+            return self.collectPaymentMethod(intent: activePaymentIntent)
+                .flatMap { intent in
+                    self.processPayment(intent: intent)
+                }
+                .map(PaymentIntent.init(intent:))
```

Comment:
This `intent:` syntax without passing anything specific into the properties always confuses me 😅 , just to clarify my understanding: 

We're mapping the received values to `PaymentIntent` objects, and each of those is created using the convenience init. Rather than depending on a specific argument, each `StripePaymentIntent` object we receive in the sequence would create a corresponding `PaymentIntent`. Is that right?

Reply:
Sorry, it was to match the existing code. Yes, your description sounds right.

This is a one-line version of the following:
```suggestion
                .map {
                    PaymentIntent(intent: $0)
                }
```

Reply:
> This is a one-line version of the following:

Ah! Much clearer, thanks 🙇